source/upgrade/upgrading-mattermost-server.rst

```diff
@@ -1,44 +1,116 @@
 Upgrading Mattermost Server
 ===========================
 
+|all-plans| |self-hosted|
+
+.. |all-plans| image:: ../images/all-plans-badge.png
+  :scale: 30
+  :target: https://mattermost.com/pricing
+  :alt: Available in Mattermost Free and Starter subscription plans.
+
+.. |self-hosted| image:: ../images/self-hosted-badge.png
+  :scale: 30
+  :target: https://mattermost.com/deploy
+  :alt: Available for Mattermost Self-Hosted deployments.
+
 In most cases you can upgrade Mattermost Server in a few minutes, but the upgrade can take longer depending on several factors, including the size and complexity of your installation, and the version that you're upgrading from.
 
+Preparing to Upgrade to the Latest Version
+------------------------------------------
+
+A Mattermost server v6.0 upgrade will run significant database schema changes that can cause a large startup time depending on the dataset size. Zero downtime won't be possible for v6.0, but depending on the efforts made during the migration process, it can be minimized to a large extent. Running queries prior to the upgrade can also save some downtime. However, some queries can still cause full table locking and require Mattermost to be in read-only mode for the duration of the query. 
+
+We recommend that you:
+
+- Set up a maintenance window outside of working hours to mitigate the migration impact. 
+- Keep a backup of your database to ensure you can load a previous database snapshot if necessary.
+- Upgrade your instance of Mattermost to a supported Extended Support Release (ESR) version before attempting to run the v6.0 upgrade.
+
 .. important::
 
-  Support for Mattermost Server v5.31 `Extended Support Release <https://docs.mattermost.com/administration/extended-support-release.html>`_ will come to the end of its life cycle on October 15, 2021. Upgrading to Mattermost Server v5.37 `Extended Support Release <https://docs.mattermost.com/administration/extended-support-release.html>`_ or later is required.
+  Support for Mattermost Server v5.31 `Extended Support Release <https://docs.mattermost.com/administration/extended-support-release.html>`__ will come to the end of its life cycle on October 15, 2021. Upgrading to Mattermost Server v5.37 Extended Support Release or later is required.
+
+Upgrading from a previous Extended Support Release to the latest Extended Support Release is supported. However, it's important to review the :doc:`important-upgrade-notes` for all intermediate versions in between to ensure you’re aware of the possible migrations that could affect your upgrade.
```

Comment:
Can we add more specific recommendations for this per the conversation in the performance channel: https://community.mattermost.com/core/pl/6m6q4bggdig5mydi1x6mrtc7xy (from the channel - it was recommended that 5.35 could be impactful to the length of time to upgrade from 5.31 to 5.37